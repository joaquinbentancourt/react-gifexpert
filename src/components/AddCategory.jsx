import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({onNewCategory}) => {
  
    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = ( event ) => {
        event.preventDefault();  // para que no refresque
        const newInputValue = inputValue.trim();
        if(newInputValue.length <= 2) return;
        onNewCategory(newInputValue);
        setInputValue('');
    }

    return (
        <form aria-label="form" onSubmit={ onSubmit }>
            <input 
                type="text"
                placeholder="Buscar gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
    );
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
};