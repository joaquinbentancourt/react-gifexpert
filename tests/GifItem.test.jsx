import { render, screen } from "@testing-library/react";
import { GifItem } from "../src/components/GifItem";

describe('Pruebas en GifItem', () => { 
    const title = 'Título';
    const url = 'http://www.img.jpg';
    test('debería de hacer match con el snapshot', () => { 
        const {container} = render(<GifItem title={ title } url={ url } />);
        expect(container).toMatchSnapshot(); 
    });
    test('debe de mostrar la imagen con el url indicado', () => { 
        render(<GifItem url={ url } title={ title } />);
        const {src, alt} = screen.getByRole('img');
        expect(src).toContain(url);
        expect(alt).toBe(title);
    });
    test('debe de mostrar el título en el componente', () => { 
        render(<GifItem url={ url } title={ title } />);
        expect(screen.getByText(title)).toBeTruthy();
    });
 });