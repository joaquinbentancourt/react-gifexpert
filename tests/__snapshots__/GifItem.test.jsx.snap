// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem debería de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Título"
      src="http://www.img.jpg"
    />
    <p>
      Título
    </p>
  </div>
</div>
`;
